addons/account/models/account_move.py

```diff
@@ -1842,34 +1881,20 @@ def create_tax_cash_basis_entry(self, percentage_before_rec):
-                                newly_created_move = self._create_tax_basis_move()
                             #create cash basis entry for the base
-                            for tax in line.tax_ids:
-                                account_id = self._get_tax_cash_basis_base_account(line, tax)
-                                self.env['account.move.line'].with_context(check_move_validity=False).create({
-                                    'name': line.name,
-                                    'debit': rounded_amt > 0 and rounded_amt or 0.0,
-                                    'credit': rounded_amt < 0 and abs(rounded_amt) or 0.0,
-                                    'account_id': account_id.id,
-                                    'tax_exigible': True,
-                                    'tax_ids': [(6, 0, [tax.id])],
-                                    'move_id': newly_created_move.id,
-                                    'currency_id': line.currency_id.id,
-                                    'amount_currency': self.amount_currency and line.currency_id.round(line.amount_currency * amount / line.balance) or 0.0,
-                                    'partner_id': line.partner_id.id,
-                                })
-                                self.env['account.move.line'].with_context(check_move_validity=False).create({
-                                    'name': line.name,
-                                    'credit': rounded_amt > 0 and rounded_amt or 0.0,
-                                    'debit': rounded_amt < 0 and abs(rounded_amt) or 0.0,
-                                    'account_id': account_id.id,
-                                    'tax_exigible': True,
-                                    'move_id': newly_created_move.id,
-                                    'currency_id': line.currency_id.id,
-                                    'amount_currency': self.amount_currency and line.currency_id.round(-line.amount_currency * amount / line.balance) or 0.0,
-                                    'partner_id': line.partner_id.id,
-                                })
+                            for tax in line.tax_ids.filtered(lambda t: t.tax_exigibility == 'on_payment'):
+                                key = self._get_tax_cash_basis_hash(tax, move, line)
+                                caba_keys.add(key)
+                                caba_amount_dict.setdefault(key, 0)
+                                caba_amount_dict[key] += rounded_amt
+                                caba_amount_currency_dict.setdefault(key, 0)
+                                caba_amount_currency_dict[key] += line.currency_id.round(line.amount_currency * amount / line.balance) if line.currency_id and self.amount_currency else 0.0
+
+            if caba_keys:
+                if not newly_created_move:
+                    newly_created_move = self._create_tax_basis_move()
+                # create cash basis entry for the base
```

Comment:
Useless comment, rather replace it by a small comment on top on the line getting the key explaining that we want to group base lines as much as possible to avoid creating too many of them.